### DISCUSSION TITLE
Filter By Sub Entity

### DISCUSSION THREAD

**Discussion by BishoyNarouz** [prisma/prisma#13942] [Q&A]
```prisma
model Lead {
    id                 Int           @id @default(autoincrement())
    email              String?       @db.VarChar(200)
    phone              String        @db.VarChar(200)
    fullName           String        @db.VarChar(200)
    leadHistory        LeadHistory[]
    @@map(name: "leads")
}

model LeadHistory {
    id            Int         @id @default(autoincrement())
    comment       String?     @db.Text
    callStartDate DateTime
    callEndDate   DateTime
    status        Status      @relation(fields: [statusId], references: [id])
    statusId      Int
    lead          Lead        @relation(fields: [leadId], references: [id])
    leadId        Int

    @@index([id])
    @@map(name: "lead_histories")
}
```
i need to get all leads which has last statusId = 3
how can i write my query?

**Comment by austincrim** [comment]
Hey @BishoyNarouz 👋,

Could you give me a little more context? 
Do you only want to select the most recent `LeadHistory` with a `statusId` of 3?

**Reply by BishoyNarouz** [reply]
I need to get all leads which has last lead history statusid is 3
If last statusid in lead history does not equal 3 do not return it
Is that clear?

**Reply by austincrim** [reply]
When you say "**last** statusId", are you referring to the most recent one determined by some date? Does the `Status` model have an `updatedAt` field?

**Reply by BishoyNarouz** [reply]
check the last entry in lead history if statusId not equal 3 do not return lead
make sense?

**Reply by austincrim** [reply]
Ok, just determining the "last entry" by the `LeadHistory` id?

I don't think this is possible in Prisma. Using [`$queryRaw`](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryraw) is probably your best bet.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
